sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3521,6 +3521,45 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
             |""".stripMargin), Row(1))
     }
   }
+
+  test("SPARK-31875: disable hint resolution when spark.sql.optimizer.ignoreHints = true") {
```

Comment:
@dilipbiswal . Can we have a more precise test case name because this PR will remove hint instead of `disable hint resolution` from now?